### Analysis Data using Panda for Mail Customer

In [13]:
import pandas as pd
import numpy as np

## 1. Loading the Dataset

In [12]:
mc = pd.read_csv('csv/Mall_Customers.csv')

In [18]:
mc_csv = np.genfromtxt('csv/Mall_Customers.csv', delimiter='',skip_header=1)
mc_csv

#Note: Notice the gender column is all NaNs. Why is that? How can you solve for it?
#The reason the "gender" column is all NaNs when using np.genfromtxt is because genfromtxt is designed for numerical data and, 
# by default, cannot handle string columns well—especially if the delimiter is not set correctly or if the file contains mixed data types (numbers and text). 
# In the code, you used delimiter='', which means whitespace, but your CSV is likely comma-separated and contains text columns.


array([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, na

In [ ]:
#To solve this:
#Use the correct delimiter: delimiter=','
#For mixed data (numbers and text), use dtype=None and encoding='utf-8' so numpy can handle strings.
mc_csv = np.genfromtxt('csv/Mall_Customers.csv', delimiter=',', dtype=None, encoding='utf-8', names=True)
mc_csv
#Alternatively, use pandas (pd.read_csv), which is better for mixed data.


array([(  1, 'Male', 19,  15, 39), (  2, 'Male', 21,  15, 81),
       (  3, 'Female', 20,  16,  6), (  4, 'Female', 23,  16, 77),
       (  5, 'Female', 31,  17, 40), (  6, 'Female', 22,  17, 76),
       (  7, 'Female', 35,  18,  6), (  8, 'Female', 23,  18, 94),
       (  9, 'Male', 64,  19,  3), ( 10, 'Female', 30,  19, 72),
       ( 11, 'Male', 67,  19, 14), ( 12, 'Female', 35,  19, 99),
       ( 13, 'Female', 58,  20, 15), ( 14, 'Female', 24,  20, 77),
       ( 15, 'Male', 37,  20, 13), ( 16, 'Male', 22,  20, 79),
       ( 17, 'Female', 35,  21, 35), ( 18, 'Male', 20,  21, 66),
       ( 19, 'Male', 52,  23, 29), ( 20, 'Female', 35,  23, 98),
       ( 21, 'Male', 35,  24, 35), ( 22, 'Male', 25,  24, 73),
       ( 23, 'Female', 46,  25,  5), ( 24, 'Male', 31,  25, 73),
       ( 25, 'Female', 54,  28, 14), ( 26, 'Male', 29,  28, 82),
       ( 27, 'Female', 45,  28, 32), ( 28, 'Male', 35,  28, 61),
       ( 29, 'Female', 40,  29, 31), ( 30, 'Female', 23,  29, 87),
       ( 31, 'Male', 

## 2. Understand Customer Demographics

In [ ]:
#2a. Calculate the average Age, Annual Income, and Spending Score to understand the typical customer profile
#Hint: Use NumPy operations like mean(), min(), and max() for this analysis.

age_mean = np.mean(mc['Age'])
income_mean = np.mean(mc['Annual Income (k$)'])
score_mean = np.mean(mc['Spending Score (1-100)'])

age_min = np.min(mc['Age'])
income_min = np.min(mc['Annual Income (k$)'])
score_min = np.min(mc['Spending Score (1-100)'])

age_max = np.max(mc['Age'])
income_max = np.max(mc['Annual Income (k$)'])
score_max = np.max(mc['Spending Score (1-100)'])

print(f"Average Age: {age_mean:.2f}, Min: {age_min}, Max: {age_max}")
print(f"Average Annual Income (k$): {income_mean:.3f}, Min: {income_min}, Max: {income_max}")
print(f"Average Spending Score: {score_mean:.2f}, Min: {score_min}, Max: {score_max}")

Average Age: 38.85, Min: 18, Max: 70
Average Annual Income (k$): 60.560, Min: 15, Max: 137
Average Spending Score: 50.20, Min: 1, Max: 99


Summary Analysis:
-   The average age and income values help you understand the typical demographic and economic status of your customers.
-   The range (min to max) for each metric shows the diversity in your customer base.
-   The spending score average and range indicate how actively customers are spending, which can help in segmenting high-value customers or identifying potential for targeted marketing.

In [46]:
#2b. Analyze customer distribution by gender. Does one gender tend to spend more or earn more on average
#using same NumPy operations like mean(), min(), and max() for this analysis

for gender in mc['Gender'].unique():
    gender_group = mc[mc['Gender'] == gender]
    age_mean = np.mean(gender_group['Age'])
    income_mean = np.mean(gender_group['Annual Income (k$)'])
    score_mean = np.mean(gender_group['Spending Score (1-100)'])
    print(f"{gender}:")
    print(f"  Average Age: {age_mean:.2f}")
    print(f"  Average Annual Income (k$): {income_mean:.2f}")
    print(f"  Average Spending Score: {score_mean:.2f}\n")

Male:
  Average Age: 39.81
  Average Annual Income (k$): 62.23
  Average Spending Score: 48.51

Female:
  Average Age: 38.10
  Average Annual Income (k$): 59.25
  Average Spending Score: 51.53



Summary Analysis:
-   Average Age: Shows if one gender group is generally younger or older.
-   Average Annual Income: Reveals if one gender tends to earn more on average.
-   Average Spending Score: Indicates if one gender tends to spend more or is more engaged as a customer.
-   If one gender has a higher average annual income, they may represent a more affluent segment.
-   If one gender has a higher average spending score, they may be more valuable or engaged customers.
-   Comparing average ages can help you tailor marketing strategies to the dominant age group within each gender.

## 3. Identify High-Value Customers

In [61]:
#3a. Filter customers with a Spending Score > 80 and calculate their average Annual Income
#Use boolean indexing and NumPy's argsort() for ranking.

#filter high-value customers for Spending Score > 80
high_value_customers = mc[mc['Spending Score (1-100)'] > 80]

# Calculate average Annual Income of high-value customers
avg_income_high_value = np.mean(high_value_customers['Annual Income (k$)'])

# Print the average Annual Income of high-value customers
print(f"Average Annual Income of High-Value Customers (Spending Score > 80): {avg_income_high_value:.2f} (k$)")

# Use argsort to get indices of high-value customers sorted by Spending Score
# the output is the indexes of the high-value customers sorted in descending order by Spending Score
#print(high_value_customers['Spending Score (1-100)'].values)

top_indices = np.argsort(-high_value_customers['Spending Score (1-100)'].values)
#print(top_indices)

# Optionally, show top 10 high-value customers by Spending Score
top_customers = high_value_customers.iloc[top_indices][:10]
print("Top 5 High-Value Customers:")
print(top_customers)

Average Annual Income of High-Value Customers (Spending Score > 80): 69.07 (k$)
Top 5 High-Value Customers:
     CustomerID  Gender  Age  Annual Income (k$)  Spending Score (1-100)
11           12  Female   35                  19                      99
19           20  Female   35                  23                      98
145         146    Male   28                  77                      97
185         186    Male   30                  99                      97
167         168  Female   33                  86                      95
127         128    Male   40                  71                      95
7             8  Female   23                  18                      94
141         142    Male   32                  75                      93
163         164  Female   31                  81                      93
173         174    Male   36                  87                      92


Summary Analysis:

Customers with a Spending Score > 80 are identified as high-value customers.
The average annual income of these high-value customers is calculated, showing that they tend to have a higher-than-average income compared to the general customer base.
The top high-value customers are sorted by their spending score, highlighting those who are the most engaged and likely to spend more.
Marketing Recommendations:

Targeted Campaigns:
Focus marketing efforts on high-value customers with personalized offers, loyalty programs, or exclusive deals, as they are more likely to respond positively and increase their spending.

Upselling and Cross-selling:
Since these customers already have high spending scores and above-average incomes, introduce premium products or bundled offers to maximize revenue from this segment.

Customer Retention:
Implement retention strategies such as VIP events, early access to sales, or special recognition to maintain engagement and loyalty among top spenders.

Profile Analysis:
Analyze the demographic and behavioral patterns of these top customers (e.g., age, gender, income) to identify similar profiles in your broader customer base for potential upsell opportunities.

Feedback and Engagement:
Engage high-value customers for feedback and involve them in product development or beta testing, as their preferences can guide future offerings and improve satisfaction.

In [62]:
#3b. Identify the top 10 customers by Spending Score. What do they have in common (e.g.,age group, gender)
#Use boolean indexing and NumPy's argsort() for ranking.

# Get indices that would sort all customers by Spending Score in descending order
top10_indices = np.argsort(-mc['Spending Score (1-100)'].values)[:10]
top10_customers = mc.iloc[top10_indices]

print("Top 10 Customers by Spending Score:")
print(top10_customers)

# Optionally, summarize what they have in common
print("\nSummary of Top 10 Customers:")
print(top10_customers[['Gender', 'Age', 'Annual Income (k$)', 'Spending Score (1-100)']].describe(include='all'))
print("\nGender distribution among top 10:")
print(top10_customers['Gender'].value_counts())


Top 10 Customers by Spending Score:
     CustomerID  Gender  Age  Annual Income (k$)  Spending Score (1-100)
11           12  Female   35                  19                      99
19           20  Female   35                  23                      98
185         186    Male   30                  99                      97
145         146    Male   28                  77                      97
127         128    Male   40                  71                      95
167         168  Female   33                  86                      95
7             8  Female   23                  18                      94
141         142    Male   32                  75                      93
163         164  Female   31                  81                      93
41           42    Male   24                  38                      92

Summary of Top 10 Customers:
        Gender        Age  Annual Income (k$)  Spending Score (1-100)
count       10  10.000000           10.000000               1

Summary Analysis:

The top 10 customers by Spending Score are identified using NumPy's argsort() to sort all customers in descending order of their spending score.
By examining their demographic and financial data (age, gender, annual income), you can observe common characteristics among these high spenders.
The summary statistics (mean, min, max, etc.) and gender distribution help reveal patterns, such as whether top spenders are predominantly of a certain gender, age group, or income bracket.
What they have in common (example insights):

If the summary shows a narrow age range, most top spenders may belong to a specific age group.
If one gender dominates, marketing can be tailored accordingly.
If their average income is higher than the overall average, it suggests that higher income correlates with higher spending scores.
Marketing Recommendations:

Targeted Promotions:
Design campaigns focused on the dominant age group or gender among top spenders.

Premium Offers:
If top spenders have higher incomes, introduce premium products or exclusive services to this segment.

Personalized Engagement:
Use personalized communication and loyalty programs to retain these valuable customers.

Lookalike Audiences:
Use the demographic profile of top spenders to identify and target similar customers in your broader base.

Feedback Collection:
Engage these top customers for feedback to further refine your offerings and customer experience.



## 4. Explore Relationships Between Features

In [ ]:
#Compute the pairwise correlations between Age, Annual Income, and Spending Score to uncover key drivers of spending
#Correlation can be computed using NumPy's corrcoef().

# Extract relevant columns as a NumPy array
data = mc[['Age', 'Annual Income (k$)', 'Spending Score (1-100)']].values

# Compute the correlation matrix
corr_matrix = np.corrcoef(data, rowvar=False)

# Display the correlation matrix with labels
columns = ['Age', 'Annual Income (k$)', 'Spending Score (1-100)']
print("Pairwise Correlation Matrix:")
print(pd.DataFrame(corr_matrix, index=columns, columns=columns))

#np.corrcoef computes the correlation matrix for the given data.
#The correlation coefficient (Pearson’s r) measures the linear relationship between two variables, 
#ranging from -1 (perfect negative) to +1 (perfect positive), with 0 meaning no linear correlation

#Why is this useful?
#High positive value: Strong direct relationship (as one increases, so does the other).
#High negative value: Strong inverse relationship (as one increases, the other decreases).
#Near zero: No linear relationship.


Pairwise Correlation Matrix:
                             Age  Annual Income (k$)  Spending Score (1-100)
Age                     1.000000           -0.012398               -0.327227
Annual Income (k$)     -0.012398            1.000000                0.009903
Spending Score (1-100) -0.327227            0.009903                1.000000


Interpretation:

Spending Score vs. Age:
Correlation = -0.327227
This is a moderate negative correlation.
Meaning: As age increases, spending score tends to decrease. Older customers generally have lower spending scores.

Spending Score vs. Annual Income:
Correlation = 0.009903
This value is very close to zero.
Meaning: There is almost no linear relationship between annual income and spending score. Customers with higher or lower income do not consistently spend more or less.

Summary:

Age is somewhat negatively related to spending score (older customers spend less).
Annual income has almost no effect on spending score (income does not predict spending behavior in this dataset).

In [65]:
#Filter young adults (18-25) and calculate their average Spending Score. Compare this with older age groups
#using Correlation can be computed using NumPy's corrcoef()

# Young adults: 18-25
young_adults = mc[(mc['Age'] >= 18) & (mc['Age'] <= 25)]
avg_score_young = np.mean(young_adults['Spending Score (1-100)'])

# Older adults: 26+
older_adults = mc[mc['Age'] > 25]
avg_score_older = np.mean(older_adults['Spending Score (1-100)'])

print(f"Average Spending Score (Age 18-25): {avg_score_young:.2f}")
print(f"Average Spending Score (Age 26+): {avg_score_older:.2f}")

Average Spending Score (Age 18-25): 54.95
Average Spending Score (Age 26+): 49.09


##### Summary: Spending Score by Age Group and Correlation Insights

- **Average Spending Score (Age 18-25):** *(output from code, e.g., 59.38)*
- **Average Spending Score (Age 26+):** *(output from code, e.g., 48.65)*

This shows that **younger adults (18-25) have a higher average spending score** compared to older adults (26+). This supports the earlier correlation analysis.

##### Related Correlation Matrix (from previous analysis):

|                        | Age      | Annual Income (k$) | Spending Score (1-100) |
|------------------------|----------|--------------------|------------------------|
| **Age**                | 1.000    | -0.012             | -0.327                 |
| **Annual Income (k$)** | -0.012   | 1.000              | 0.010                  |
| **Spending Score**     | -0.327   | 0.010              | 1.000                  |

- **Spending Score vs. Age:** Correlation = **-0.33** (moderate negative)
    - As age increases, spending score tends to decrease.
    - The group comparison above confirms this trend: **older customers generally have lower spending scores**.

- **Spending Score vs. Annual Income:** Correlation = **0.01** (almost no relationship)
    - Income does not predict spending score in this dataset.

**Conclusion:**  
The data and correlation matrix both indicate that age is a key factor influencing spending score, with younger customers being more active spenders.

## 5. Customer Segmentation

In [68]:
# Customer Segmentation by Spending Score

#Low: 1-33
low_segment = mc[(mc['Spending Score (1-100)'] >= 1) & (mc['Spending Score (1-100)'] <= 33)]
#Medium: 34-66
medium_segment = mc[(mc['Spending Score (1-100)'] >= 34) & (mc['Spending Score (1-100)'] <= 66)]
#High: 67-100
high_segment = mc[(mc['Spending Score (1-100)'] >= 67) & (mc['Spending Score (1-100)'] <= 100)]

# Compute averages
segments = {
    'Low': low_segment,
    'Medium': medium_segment,
    'High': high_segment
}

for name, group in segments.items():
    avg_age = np.mean(group['Age'])
    avg_income = np.mean(group['Annual Income (k$)'])
    print(f"{name} Segment:")
    print(f"  Average Age: {avg_age:.2f}")
    print(f"  Average Annual Income (k$): {avg_income:.2f}")
    print(f"  Number of Customers: {len(group)}\n")

Low Segment:
  Average Age: 42.88
  Average Annual Income (k$): 67.00
  Number of Customers: 49

Medium Segment:
  Average Age: 42.01
  Average Annual Income (k$): 53.86
  Number of Customers: 94

High Segment:
  Average Age: 30.18
  Average Annual Income (k$): 66.07
  Number of Customers: 57



##### Segment Characteristics & Marketing Strategies
**Low Segment (Spending Score 1-33)**
- *Characteristics:*
  - Typically older customers or those less engaged with the store.
  - May have a wide range of annual incomes, but low spending activity.
- *Marketing Strategies:*
  - Run awareness campaigns to increase engagement.
  - Offer introductory discounts or basic loyalty programs.
  - Educate about products/services to spark interest.

**Medium Segment (Spending Score 34-66)**
- *Characteristics:*
  - Represent the average customer base.
  - Moderate age and income levels.
  - Show some engagement but not highly active.
- *Marketing Strategies:*
  - Use personalized promotions and targeted offers.
  - Provide incentives (e.g., points, small rewards) to encourage more frequent purchases.
  - Regularly engage through newsletters or events to move them to the high segment.

**High Segment (Spending Score 67-100)**
- *Characteristics:*
  - Often younger and/or highly engaged customers.
  - May have moderate to high annual incomes.
  - Most active and valuable segment for the business.
- *Marketing Strategies:*
  - Implement VIP or premium loyalty programs.
  - Offer exclusive deals, early access to new products, or special events.
  - Focus on retention with personalized communication and recognition.

**Summary:** 
Segmenting customers by spending score helps tailor marketing strategies:  
- Low: Focus on activation and education.  
- Medium: Encourage higher engagement.  
- High: Reward loyalty and maximize retention.

# 6. Find Behavioral Similarities

In [69]:
# Normalize the dataset (subtract the mean and divide by the standard deviation for each column
# Use NumPy's vectorized operations for distance calculations

features = ['Age', 'Annual Income (k$)', 'Spending Score (1-100)']
data = mc[features].values

# Compute mean and std for each column
means = np.mean(data, axis=0)
stds = np.std(data, axis=0)

# Normalize
normalized_data = (data - means) / stds

print("First 5 rows of normalized data:")
print(normalized_data[:5])

# Example: Compute Euclidean distance between first two customers
dist = np.linalg.norm(normalized_data[0] - normalized_data[1])
print(f"Euclidean distance between first two customers: {dist:.2f}")

First 5 rows of normalized data:
[[-1.42456879 -1.73899919 -0.43480148]
 [-1.28103541 -1.73899919  1.19570407]
 [-1.3528021  -1.70082976 -1.71591298]
 [-1.13750203 -1.70082976  1.04041783]
 [-0.56336851 -1.66266033 -0.39597992]]
Euclidean distance between first two customers: 1.64


In [70]:
#Compute the Euclidean distance between each customer and the customer with the highest Spending Score
#Use NumPy's vectorized operations for distance calculations

# Find the index of the customer with the highest Spending Score
max_score_idx = mc['Spending Score (1-100)'].idxmax()

# Get the normalized feature vector for that customer
target_vector = normalized_data[max_score_idx]

# Compute Euclidean distances from all customers to the target customer (vectorized)
distances = np.linalg.norm(normalized_data - target_vector, axis=1)

# Show the first 5 distances as an example
print("First 5 distances to the highest Spending Score customer:")
print(distances[:5])

First 5 distances to the highest Spending Score customer:
[2.60142971 1.23333086 3.76921616 1.21828611 2.30965307]


In [71]:
#Identify customers who exhibit similar behavior
#Use NumPy's vectorized operations for distance calculations

# Let's find the 5 customers most similar to the highest Spending Score customer (excluding themselves)
similar_indices = np.argsort(distances)[1:6]  # [0] is the customer themselves

print("Top 5 customers most similar to the highest Spending Score customer:")
print(mc.iloc[similar_indices][['Gender', 'Age', 'Annual Income (k$)', 'Spending Score (1-100)']])

Top 5 customers most similar to the highest Spending Score customer:
    Gender  Age  Annual Income (k$)  Spending Score (1-100)
19  Female   35                  23                      98
25    Male   29                  28                      82
7   Female   23                  18                      94
29  Female   23                  29                      87
23    Male   31                  25                      73


# Typical Analyst Deliverable

## Mall Customer Analysis Report

This report summarizes key findings from the analysis of mall customer data, focusing on demographics, spending patterns, high-value customers, customer segments, and the relationships between age, income, and spending.

---

#### 1. Customer Demographics

- **Average Age:** The typical customer is around the dataset's mean age, with a wide age range.
- **Average Annual Income:** Customers have diverse income levels, with the mean providing a central tendency.
- **Average Spending Score:** The average spending score indicates moderate engagement, with some customers spending much more or less than average.
- **Gender Distribution:** Both genders are represented, with minor differences in average age, income, and spending score.

**Key Finding:**  
The customer base is diverse in age and income, with both genders showing similar spending patterns.

---

#### 2. High-Value Customers

- **Definition:** Customers with a Spending Score > 80.
- **Characteristics:**
  - Tend to have higher-than-average annual incomes.
  - Are among the most engaged and active spenders.
- **Top Customers:** The top 10 customers by spending score often share similar demographic traits (e.g., age group, sometimes gender).

**Key Finding:**  
High-value customers are not only high spenders but also tend to have higher incomes, making them ideal targets for premium offers and loyalty programs.

---

#### 3. Customer Segmentation

Customers were grouped by Spending Score:

| Segment | Spending Score Range | Average Age | Average Income (k$) | Characteristics | Marketing Strategies |
|---------|---------------------|-------------|---------------------|-----------------|---------------------|
| Low     | 1-33                | Older       | Varies              | Less engaged    | Awareness campaigns, basic loyalty, education |
| Medium  | 34-66               | Moderate    | Moderate            | Average         | Personalized offers, incentives, regular engagement |
| High    | 67-100              | Younger     | Moderate-High       | Highly engaged  | VIP programs, exclusive deals, retention focus |

**Key Finding:**  
Segmenting by spending score reveals actionable groups for targeted marketing, with the high segment being the most valuable.

---

#### 4. Relationships Between Features

- **Age vs. Spending Score:**  
  - **Correlation:** Moderate negative (~-0.33)
  - **Insight:** Younger customers tend to have higher spending scores; older customers spend less.
- **Income vs. Spending Score:**  
  - **Correlation:** Near zero
  - **Insight:** Annual income does not predict spending score; high earners are not always high spenders.
- **Age vs. Income:**  
  - **Correlation:** Near zero
  - **Insight:** No strong relationship between age and income in this dataset.

**Key Finding:**  
Age is a more important driver of spending behavior than income.

---

#### 5. Behavioral Similarities

- **Method:** Normalized features and computed Euclidean distances.
- **Finding:** Customers most similar to the highest spender can be identified for targeted campaigns or lookalike modeling.

---

#### 6. Business Recommendations

- **For High-Value Customers:**  
  Focus on retention, VIP programs, and exclusive offers.
- **For Low-Value Customers:**  
  Increase engagement through education and introductory promotions.
- **For Medium Segment:**  
  Use incentives and regular engagement to move them to the high segment.
- **General:**  
  Target younger customers for high-engagement campaigns; income-based targeting is less effective.

---

**Summary:**  
The analysis reveals that age is a key factor in spending, high-value customers are both high earners and spenders, and segmentation enables tailored marketing strategies. Focusing on customer engagement, especially among younger demographics, will likely yield the best business results.



## Actionable Recommendations

#### 1. Discounts and Engagement for Low-Spending Customers
- **Strategy:** Offer time-limited discounts, welcome vouchers, or introductory bundles to customers in the low spending segment (Spending Score 1-33).
- **Goal:** Encourage more frequent visits and increase initial engagement.
- **Tactics:** Send personalized emails or app notifications highlighting special offers for first-time or infrequent shoppers.

#### 2. Exclusive Loyalty Programs for High-Spending Customers
- **Strategy:** Create a VIP loyalty program for high-value customers (Spending Score 67-100), offering exclusive rewards, early access to sales, and invitations to special events.
- **Goal:** Retain top spenders and increase their lifetime value.
- **Tactics:** Use personalized communication (e.g., thank-you notes, birthday gifts) and provide tailored recommendations based on purchase history.

#### 3. Targeted Campaigns by Age Group
- **Strategy:** Use targeted advertising and product recommendations for younger adults (18-25), who have higher spending scores, and for older adults (26+), who may need different incentives.
- **Goal:** Maximize marketing ROI by aligning offers with the preferences of each age group.
- **Tactics:** Promote trendy or tech-focused products to younger customers via social media, and highlight value or convenience for older segments through email or in-store promotions.

---
**Summary:**  
Personalized marketing—using discounts, loyalty programs, and targeted campaigns—will help activate low spenders, retain high-value customers, and optimize engagement across different age groups.
